dc-cudami-server/dc-cudami-server-backend-jdbi/src/main/java/de/digitalcollections/cudami/server/backend/impl/jdbi/identifiable/entity/ProjectRepositoryImpl.java

```diff
@@ -124,25 +127,37 @@ public boolean addDigitalObjects(UUID projectUuid, List<DigitalObject> digitalOb
             + doTableAlias
             + ".uuid = pd.digitalobject_uuid"
             + " WHERE pd.project_uuid = :uuid";
+    Map<String, Object> argumentMappings = new HashMap<>();
+    argumentMappings.put("uuid", projectUuid);
+
+    String searchTerm = searchPageRequest.getQuery();
+    if (StringUtils.hasText(searchTerm)) {
+      commonSql += " AND " + getCommonSearchSql(doTableAlias);
+      argumentMappings.put("searchTerm", this.escapeTermForJsonpath(searchTerm));
+    }
 
     StringBuilder innerQuery = new StringBuilder("SELECT pd.sortindex AS idx, *" + commonSql);
-    addFiltering(pageRequest, innerQuery);
-    pageRequest.setSorting(null);
-    innerQuery.append(" ORDER BY idx ASC");
-    addPageRequestParams(pageRequest, innerQuery);
+    addFiltering(searchPageRequest, innerQuery);
+
+    String orderBy = null;
+    if (searchPageRequest.getSorting() == null) {
+      orderBy = "ORDER BY idx ASC";
+      innerQuery.append(" ").append(orderBy);
+    }
```

Comment:
Könnte man das nicht in einer Basisklasse unterbringen? Als (static) Methode oder in `<repoImpl>:retrieveList`... Das wiederholt sich ja schon viel zu oft.